src/app/booking/OOSPlannerSection.js

```diff
@@ -183,23 +183,27 @@ const StyledTableCell = styled(TableCell)(({ theme }) => ({
 }));
 
 function FamilyBookings() {
-  const today = dayjs();
+  const { selectedDate, setSelectedDate } = useContext(SelectedDateContext);
 
-  let currentWeekFirstDayDate;
-  if (today.weekday() === 0 || today.weekday() === 6) {
-    currentWeekFirstDayDate = today.startOf("week").add(1, "day");
-  } else {
-    currentWeekFirstDayDate = today.startOf("week").add(1, "day");
-  }
-  const [weekFirstDayDate, setWeekFirstDayDate] = useState(currentWeekFirstDayDate);
-  const weekDates = Array.from({ length: 5 }, (_, i) => weekFirstDayDate.add(i, "day"));
+  const getStartOfWeek = date => {
+    let day = date.startOf('week').add(1, 'day');
+    if (date.weekday() === 6) {
+      day = date.add(2, 'days').startOf('week').add(1, 'day');
+    } else if (date.weekday() === 0) {
+      day = date.add(1, 'day').startOf('week').add(1, 'day');
+    }
+    return day;
+  };
+
+  const weekFirstDayDate = getStartOfWeek(selectedDate);
+  const weekDates = Array.from({ length: 5 }, (_, i) => weekFirstDayDate.add(i, 'day'));
   const { data: children } = useQuery("children", getChildren);
   const { data: bookings } = useQuery("bookings", () => getBookings(weekDates[0], weekDates[4]));
 
   const formatDate = (date) => date.format("ddd D");
 
-  const handleNextWeek = () => setWeekFirstDayDate(weekFirstDayDate.add(7, "day"));
-  const handlePreviosWeek = () => setWeekFirstDayDate(weekFirstDayDate.subtract(7, "day"));
+  const handleNextWeek = () => setSelectedDate(weekFirstDayDate.add(7, "days"));
```

Comment:
days is not a valid argument for [dayjs().add()](https://day.js.org/docs/en/manipulate/add)